pkg/build/strategies/sti/sti.go

```diff
@@ -198,6 +221,47 @@ func (builder *STI) Prepare(config *api.Config) error {
 		WorkingDir: config.WorkingDir,
 	}
 
+	if len(config.RuntimeImage) > 0 {
+		if config.Incremental {
+			return fmt.Errorf("Incremental build with runtime image isn't supported")
```

Comment:
But in this case it wouldn't fail for invocation inside of OpenShift, right? I tried to catch a both cases and I didn't find a better place for that.

Are you still thinking that this check should be moved to `main.go`?